validator/sawtooth_validator/networking/future.py

```diff
@@ -133,18 +133,20 @@ def remove(self, correlation_id):
                 "no such correlation id: {}".format(correlation_id))
 
     def remove_expired(self):
-        correlation_ids = [
-            correlation_id
-            for correlation_id, future
-            in self._futures.items()
-            if future.is_expired()
-        ]
-        for correlation_id in correlation_ids:
-            self._futures[correlation_id].timer_stop()
-            del self._futures[correlation_id]
+        with self._lock:
```

Comment:
The commit message is confusing. `resolve` is not a method. You updated the remove_expired function. Can you redo the commit message.